nx/lib/nx.ex

```diff
@@ -7046,6 +7046,123 @@ defmodule Nx do
     impl!(tensor).sort(tensor, tensor, axis: axis, comparator: comparator)
   end
 
+  @doc """
+  Calculates the QR decomposition of a 2-D tensor with shape `{M, N}`.
+
+  ## Options
+
+  `:mode` - Can be one of `:reduced`, `:complete`. Defaults to `:reduced`
+    For the following, `K = min(M, N)`
+
+    * `:reduced` - returns `q` and `r` with shapes `{M, K}` and `{K, N}`
+    * `:complete` - returns `q` and `r` with shapes `{M, M}` and `{M, N}`
+
+  `:q_names` - Defines the axes for the `q` tensor
+
+  `:r_names` - Defines the axes for the `r` tensor
+
+  ## Examples
+
+  Note: The following examples round elements of the output tensors
```

Comment:
Sorry I missed this in the first, one I saw you moved the test so you can remove this!

Reply:
Done!